File: TsinghuaNet.Eto/TsinghuaNet.Eto/Views/SettingsDialog.xeto.vb
Imports Eto.Forms
Imports Eto.Serialization.Xaml
Imports TsinghuaNet.Models

Public Class SettingsDialog
    Inherits Dialog

    Public ReadOnly Property Packages As IReadOnlyList(Of PackageBox) = New List(Of PackageBox) From
    {
        New PackageBox("Eto.Forms", "BSD-3许可证"),
        New PackageBox("Eto.Platform.Gtk", "BSD-3许可证"),
        New PackageBox("Eto.Platform.Mac64", "BSD-3许可证"),
        New PackageBox("Eto.Platform.Wpf", "BSD-3许可证"),
        New PackageBox("Eto.Serialization.Xaml", "BSD-3许可证"),
        New PackageBox("HtmlAgilityPack", "MIT许可证"),
        New PackageBox("Newtonsoft.Json", "MIT许可证"),
        New PackageBox("Refractored.MvvmHelpers", "MIT许可证")
    }

    Public Sub New(model As MainViewModel, Optional page As Integer = 0)
        XamlReader.Load(Me)
        DataContext = model
        FindChild(Of TabControl)("SettingsTab").SelectedIndex = page
        FindChild(Of GridView)("PackageView").DataStore = Packages
    End Sub
End Class

File: TsinghuaNet.Eto/TsinghuaNet.Eto/Views/SettingsDialog.xeto.cs
using System;
using System.Diagnostics;
using Eto;
using Eto.Forms;
using Eto.Serialization.Xaml;
using TsinghuaNet.Eto.Models;
using TsinghuaNet.Eto.ViewModels;

namespace TsinghuaNet.Eto.Views
{
    public class SettingsDialog : Dialog
    {
        private SettingsViewModel Model;

#pragma warning disable 0649
        private TabControl SettingsTab;
#pragma warning restore 0649

        public SettingsDialog(int page)
        {
            XamlReader.Load(this);
            DataContext = Model = new SettingsViewModel();
            SettingsTab.SelectedIndex = page;
        }

        private const string RepoUri = "https://github.com/Berrysoft/TsinghuaNet";
        private void ShowRepo(object sender, EventArgs e)
        {
            try
            {
                var os = EtoEnvironment.Platform;
                if (os.IsWindows)
                    Process.Start(new ProcessStartInfo(RepoUri) { UseShellExecute = true });
                else if (os.IsLinux)
                    Process.Start("xdg-open", RepoUri);
                else if (os.IsMac)
                    Process.Start("open", RepoUri);
            }
            catch (Exception ex)
            {
                Debug.WriteLine(ex.ToString());
            }
        }

        private void OpenSettingsFile(object sender, EventArgs e)
        {
            try
            {
                var path = SettingsHelper.Helper.FilePath;
                var os = EtoEnvironment.Platform;
                if (os.IsWindows)
                    Process.Start(new ProcessStartInfo(path) { UseShellExecute = true });
                else if (os.IsLinux)
                    Process.Start("xdg-open", path);
                else if (os.IsMac)
                    Process.Start("open", path);
            }
            catch (Exception ex)
            {
                Debug.WriteLine(ex.ToString());
            }
        }
    }
}
